calblueprint/dccentralkitchen
# Upgrade to React Navigation 5.0

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
Refactored code for upgrade to React Navigation 5.0
- Separated stack navigators into individual files under `stack_navigators` folder
- Removed `RootStack` and moved `RewardsOverlay` modal to `StoresStack`

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links
#52 

### Online sources
[The entire API for React Navigation 5.0](https://reactnavigation.org/docs/upgrading-from-4.x)

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs
N/A

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
Play around with the app and let me know if there are any bugs/noticeable differences from master!
- Make sure the rewards modal still transitions up and down on top of the map screen
- Check that the drawer has the correct username and that the drawer items are accurate
- No screens should be erroring out

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps
Upgrade to Expo SDK 37

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases
Played around with basic functionality of the app to make sure the refactoring didn't break anything (but let me know if there's something you catch).

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots
N/A

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @anniero98 @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."
